strr-host-pm-web/app/stores/hostApplication.ts

```diff
@@ -1,55 +1,96 @@
-    specifiedServiceProvider: optionalOrEmptyString,
-    agreedToSubmit: z.boolean()
+  const { t } = useI18n()
+  const { postApplication } = useStrrApi()
+
+  const propertyStore = useHostPropertyStore()
+  const reqStore = usePropertyReqStore()
+  const ownerStore = useHostOwnerStore()
+  const documentStore = useDocumentStore()
+  const confirmationSchema = z.object({
+    agreedToRentalAct: z.boolean().refine(val => val, { message: t('validation.required') }),
+    agreedToSubmit: z.boolean().refine(val => val, { message: t('validation.required') })
   })
 
-  const getEmptyResidenceDetails = () => ({
-    isPrincipalResidence: undefined,
-    agreedToRentalAct: undefined,
-    nonPrincipalOption: '',
-    specifiedServiceProvider: '',
-    agreedToSubmit: undefined
+  const getEmptyConfirmation = () => ({
+    agreedToRentalAct: false,
+    agreedToSubmit: false
   })
 
-  const residenceDetails = ref<ApiResidence>(getEmptyResidenceDetails())
-  const documents = ref<ApiDocument[]>([])
+  const userConfirmation = ref(getEmptyConfirmation())
+
+  const validateUserConfirmation = (returnBool = false): MultiFormValidationResult | boolean => {
+    const result = validateSchemaAgainstState(confirmationSchema, userConfirmation.value, 'confirmation-form')
+
+    if (returnBool) {
+      return result.success === true
+    } else {
+      return [result]
+    }
+  }
+
+  const createApplicationBody = (): HostApplicationPayload => {
+    const host = ownerStore.findByRole(OwnerRole.HOST)
+    const cohost = ownerStore.findByRole(OwnerRole.CO_HOST)
+    const propertyManger = ownerStore.findByRole(OwnerRole.PROPERTY_MANAGER)
 
-  // TODO: add document schema?
-  // TODO: add document - takes file upload, post to api, push resp to documents ref
-  const removeDocumentAtIndex = (index: number) => {
-    documents.value.splice(index, 1)
+    return {
+      registration: {
+        registrationType: ApplicationType.HOST,
+        primaryContact: formatOwnerHostAPI(host as HostOwner),
+        ...(cohost
+          ? { secondaryContact: formatOwnerHostAPI(cohost) }
+          : {}
+        ),
+        ...(propertyManger
+          ? { propertyManager: formatOwnerPropertyManagerAPI(propertyManger) }
+          : {}
+        ),
+        unitAddress: formatHostUnitAddressApi(propertyStore.unitAddress.address),
+        unitDetails: formatHostUnitDetailsAPI(propertyStore.unitDetails, propertyStore.blInfo),
+        documents: documentStore.apiDocuments,
+        principalResidence: {
+          // TODO: confirm mapping - assuming this is if PR is required or not?
+          isPrincipalResidence: !reqStore.prRequirements.isPropertyPrExempt,
+          nonPrincipalOption: reqStore.prRequirements.prExemptionReason || '',
+          agreedToRentalAct: userConfirmation.value.agreedToRentalAct,
+          agreedToSubmit: userConfirmation.value.agreedToSubmit,
+          // TODO: No idea what this is supposed to be?
+          specifiedServiceProvider: ''
```

Comment:
This was the 'other service provider' option from before.

If you selected 'other' for the reason youre pr exempt, you had to give the other reason. We don't have other as an exemption value or the `specifiedServiceProvider` in the new design.